File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 21/07/2026, 23:12:53
class Solution {
    public void moveZeroes(int[] nums) {
        int j=0;
        int n = nums.length;
        int i=0;
        while(i<n && j<n){
            if(nums[i]==0){
                i++;
               
            } 
            else {
                int temp = nums[i];
                nums[i]=nums[j];
                nums[j]=temp;
                j++;
                i++;
            }
        }
    }
}

File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 21/07/2026, 23:12:59
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
       int j=0;
       for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                int temp=nums[i];
                nums[i]=nums[j];
                nums[j]=temp;
                j++;
            }
       }
    }
};